File: src/main/java/com/mgrunt/movies/services/WatchlistService.java
package com.mgrunt.movies.services;

import com.mgrunt.movies.domain.dtos.movie.WatchlistMovieResponse;
import com.mgrunt.movies.domain.dtos.watchlist.WatchlistStatusDto;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import java.util.UUID;

public interface WatchlistService {

    Page<WatchlistMovieResponse> getMoviesToWatch(UUID userId, Pageable pageable);
    Page<WatchlistMovieResponse> getMoviesWatched(UUID userId, Pageable pageable);
    WatchlistStatusDto toggleMovie(Long tmdbId, String listType, UUID userId);
}

File: frontend/movies-app/src/Services/watchlistService.js
import api from "../api/axiosConfig";

export const watchlistService = {
    async getMoviesToWatch({ page = 0, size = 10 } = {}) {
        try {
            const response = await api.get("/v1/users/watchlist/to-watch", {
                params: { page, size },
            });
            return response.data;
        } catch (error) {
            console.error("Error fetching movies to watch:", error);
            throw error;
        }
    },

    async getMoviesWatched({ page = 0, size = 10 } = {}) {
        try {
            const response = await api.get("/v1/users/watchlist/watched", {
                params: { page, size },
            });
            return response.data;
        } catch (error) {
            console.error("Error fetching watched movies:", error);
            throw error;
        }
    },
};